stock_production_lot_active/tests/test_lot_active.py

```diff
@@ -0,0 +1,34 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import ValidationError
+from odoo.tests.common import SavepointCase
+
+
+class TestLotActive(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product = cls.env.ref("product.product_product_16")
+        cls.product.write({"tracking": "lot"})
+        cls.warehouse = cls.env.ref("stock.warehouse0")
+
+    def test_duplicate_inactive_lot(self):
+        # it should not be possible to create a new lot with the same name and company
+        # even when the first lot is inactive
+        with self.assertRaises(ValidationError):
+            self.env["stock.production.lot"].create(
+                {
+                    "name": "lot1",
```

Comment:
Make sure the lot name is unique:
```suggestion
                    "name": "stock_production_lot_active lot",
```